File: app/src/main/java/com/example/foofgeek/MainRecipeAdapter.java
package com.example.foofgeek;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.foofgeek.Helper.ImageHelper;
import com.example.foofgeek.Model.RecipeItem;
import java.util.ArrayList;

public class MainRecipeAdapter extends BaseAdapter {

    private LayoutInflater inflater;
    private ArrayList<RecipeItem> recipeList;
    private int layout;
    private ImageHelper imageHelper = new ImageHelper();

    public MainRecipeAdapter(Context context, ArrayList<RecipeItem> recipeList, int layout) {
        this.inflater=(LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        this.recipeList = recipeList;
        this.layout = layout;
    }

    @Override
    public int getCount() {
        return recipeList.size();
    }

    @Override
    public Object getItem(int position) {
        return recipeList.get(position).get_recipeName();
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(convertView==null){
            convertView=inflater.inflate(layout,parent,false);
        }

        RecipeItem recipeItem = recipeList.get(position);
        ImageView icon= convertView.findViewById(R.id.img_mainListItem);
        icon.setImageBitmap(imageHelper.getBitmapFromByteArray(recipeItem.get_thumbnail()));

        TextView name= convertView.findViewById(R.id.txt_mainListItem);
        name.setText(recipeItem.get_recipeName());
        return convertView;
    }
}

File: FoodGeek1/app/src/main/java/com/example/foofgeek/MainRecipeAdapter.kt
package com.example.foofgeek

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.foofgeek.Helper.ImageHelper
import com.example.foofgeek.Model.RecipeItem

class MainRecipeAdapter(context: Context?, recipeList: java.util.ArrayList<RecipeItem>?, layout: Int) :
    BaseAdapter() {
    private val inflater: LayoutInflater
    private val recipeList: ArrayList<RecipeItem>?
    private val layout: Int
    private val imageHelper = ImageHelper()

    init {
        inflater = context!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        this.recipeList = recipeList
        this.layout = layout
    }

    override fun getCount(): Int {
        return recipeList!!.size
    }

    override fun getItem(position: Int): Any {
        return recipeList!![position]!!.get_recipeName()
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        var convertView = convertView
        if (convertView == null) {
            convertView = inflater.inflate(layout, parent, false)
        }
        val recipeItem = recipeList!![position]
        val icon = convertView.findViewById<ImageView>(R.id.img_mainListItem)
        icon.setImageBitmap(imageHelper.getBitmapFromByteArray(recipeItem!!.get_thumbnail()))
        val name = convertView.findViewById<TextView>(R.id.txt_mainListItem)
        name.text = recipeItem!!._recipeNam
        return convertView
    }
}